server/game/gameSystems/DamageSystem.ts

```diff
@@ -83,7 +83,14 @@ export class DamageSystem<TContext extends AbilityContext = AbilityContext, TPro
 
     private getDamageAmountFromEvent(event: any): number {
         if (event.amount != null) {
-            return event.amount;
+            switch (typeof event.amount) {
```

Comment:
I think it was set up this way in the reference I gave you as well, which was an oversight on our part. The tl;dr is this should go in `addPropertiesToEvent`, detailed explanation below if you're interested. We should do the same for Heal just to be consistent.

Basically there are some replacement effects that could check the damage amount before it reaches resolution, i.e. Boba Fett's Armor. For that to work we need to determine the concrete amount of damage when the event is actually created, which is when `addPropertiesToEvent` is called. If you're interested you can see the event resolution steps in `EventWindow.ts`.

Overall the event creation / resolution flow is kind of disjointed and I want to take some time to clean it up post-release, so apologies in advance for any further onboarding pains related to that

Reply:
Don't worry about it, just gets me more familiar with the code. I'll take a look at this now.